### PR TITLE
Feature/pluggable identity

### PR DESCRIPTION
**Pull request by zhelezkov**

### CODE REVIEW

File: packages/identity-dialect-dapps/src/identity-dialect-dapps.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ReverseIdentity,
+} from '@dialectlabs/sdk';
+import { PublicKey } from '@solana/web3.js';
+import { Dapp, fetchAllDapps } from './api';
+
+const NAME = 'DIALECT_DAPPS';
+
+export class DialectDappsIdentityResolver implements IdentityResolver {
+  // public key to identity
+  dappsCached?: Record<string, Identity>;
```
Comment (tsmbl) [inline packages/identity-dialect-dapps/src/identity-dialect-dapps.ts:13]: Should we periodically re-cache this?

File: packages/sdk/src/identity/identity.interface.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import type { PublicKey } from '@solana/web3.js';
+
+export interface Identity {
+  identityName: string;
```
Comment (tsmbl) [inline packages/sdk/src/identity/identity.interface.ts:4]: May be type or source would be better naming? up to you

File: packages/sdk/src/identity/identity.interface.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  publicKey: PublicKey;
+  name?: string;
```
Comment (tsmbl) [inline packages/sdk/src/identity/identity.interface.ts:6]: Should this be always defined?

File: packages/sdk/src/identity/identity.interface.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+export interface ReverseIdentity {
+  name: string;
+  identityName: string;
+  publicKey: PublicKey;
+}
+
+export interface IdentityResolver {
+  resolve(
+    publicKey: PublicKey,
+    onProgress?: (identity: Identity) => void,
```
Comment (tsmbl) [inline packages/sdk/src/identity/identity.interface.ts:23]: Do we need this?

**Review by tsmbl — CHANGES_REQUESTED** [review]
Overall great job! Several general comments
1. Please, document fetch ployfill requirement for node and working process with yarn workspaces in new repo structure
2. Let's move error handling from providers to strategies in sdk
3. Harmonize data-structure in identity provider API: I would expect to have the same data structure for both lookups

File: packages/identity-dialect-dapps/src/identity-dialect-dapps.ts (hunk 1 above)
Comment (zhelezkov) [reply packages/identity-dialect-dapps/src/identity-dialect-dapps.ts:13]: I don't think so, dapps are rarely to change, and in the frontend use-case page refresh will re-cache this basically

File: packages/sdk/src/identity/identity.interface.ts (hunk 2 above)
Comment (zhelezkov) [reply packages/sdk/src/identity/identity.interface.ts]: agree, changed to `type`

File: packages/sdk/src/identity/identity.interface.ts (hunk 4 above)
Comment (zhelezkov) [reply packages/sdk/src/identity/identity.interface.ts]: prob no, removed

File: packages/sdk/src/identity/identity.interface.ts (hunk 3 above)
Comment (zhelezkov) [reply packages/sdk/src/identity/identity.interface.ts]: right

**Pull request merged**